doc/Language/typesystem.pod6

```diff
@@ -536,17 +536,21 @@ attributes is not yet implemented. Please note that adding a multi candidate
 that differs only in its named parameters will add that candidate behind the
 already defined one and as such it won't be picked by the dispatcher.
 
-=head3 Versioning and authorship
-
-Versioning and authorship can be applied via the adverbs
-X«C«:ver<>»|:ver<> (class)» and X«C«:auth<>»|:auth<> (class)».
-Both take a string as argument, for C<:ver> the string is converted to a
-L<Version|/type/Version> object. To query a class version and author use
-C<.^ver> and C<^.auth>.
-
-    class C:ver<4.2.3>:auth<me@here.local> {}
-    say [C.^ver, C.^auth];
-    # OUTPUT: «[v4.2.3 me@here.local]␤»
+=head3 Versioning, authorship, and API
+
+Versioning, authorship, and API can be applied via the adverbs
+X«C«:ver<>»|:ver<> (class)», X«C«:auth<>»|:auth<> (class)», and
+X«C«:api<>»|:api<> (class)» respectively. All of them take a string as argument;
+for C<:ver> the string is converted to a L<Version|/type/Version> object,
+and for C<:api> the string is converted into the allomorph
```

Comment:
If true this is incorrect behvior. `:api` should act like a version to provide expected sorting behavior, and indeed this is how it is treated in CUR land.

Reply:
From what I gather, `:api` doesn't use a `Version` object like  `:ver` does:

```
class Versioned:ver<0.0.1>:auth<github:Kaiepi>:api<1> { }
 
given Versioned {
    say .^ver.^name;  # Version
    say .^auth.^name; # Str
    say .^api.^name;  # IntStr
}
```

Reply:
Ah. Well, thats a bug...  `:api` should be a version. Wish that change had been submitted for a review so the reasoning could have been clarified for everyone.

Reply:
Why should `:api` be a `Version`?  An `identifier` I could understand.  But why a `Version`?

Reply:
An API value in my book, is like an "auth" value.  At allows you to select a specific API.  There should not be any sorting on API, like there is no sorting on "auth".

Reply:
The only reason `auth` exists is to have a non-sortable trait to influence module loading. `api` is a version because in practice it is used as and expected to sort the same way as a version. I would ask what value allowing `api` to be anything but a version allows... any string you want to use as an identifier can be turned into a version, and it will sort in a consistent way when users don't use `use Foo:api<...>` (almost always).

`auth` being non-sortable is what makes it special, and yes it can make things not-consistent. But its an escape hatch -- its not meant to encourage further inconsistency.

Reply:
I can see the merit in both sides of the argument, but unfortunately, getting to the point, @uzluisf was right to start with, it's an allomorph, so the text is correct and I'm resolving this conversation.

Reply:
What exactly do you gain by marking this conversation as resolved immediately instead of waiting an extra couple days? My original comment even outlines the reason why rushing to merge stuff is LTA

Reply:
Please feel free to continue the conversation here or wherever you want. Unfortunately, from the point of view of the context, the matter is settled. The text says that api is an allomorph, you said that it's wrong if true, but @uzluisf has proved that it's true, so the text is correct in the way it's written now and the matter, strictly within the boundaries of the documentation and this PR, is setteld.
I'm not rushing to merge because there are outstanding issues. But unless what says @uzluisf here is wrong, I don't see any further problem with this sentence.